src/time_steppers.jl

```diff
@@ -63,23 +63,40 @@ function time_step!(model::Model{A}, Nt, Δt) where A <: Architecture
     Gⁿ = G.Gu.data, G.Gv.data, G.Gw.data, G.GT.data, G.GS.data
     G⁻ = Gp.Gu.data, Gp.Gv.data, Gp.Gw.data, Gp.GT.data, Gp.GS.data
 
-    Bx, By, Bz = floor(Int, Nx/Tx), floor(Int, Ny/Ty), Nz  # Blocks in grid
+    # dynamic launch configuration
+    function get_config(dims)
+        return (kernel) -> begin
+            fun = kernel.fun
+            config = launch_configuration(fun)
+
+            # adapt the suggested config from 1D to the requested grid dimensions
+            threads = floor(Int, sqrt(config.threads))
+            blocks = ceil.(Int, [Nx,Ny] ./ threads)
+            if dims == 3
```

Comment:
@ali-ramadhan this is the messy part I was talking about, kept it like this so that I didn't need to change the actual kernels, but can of course be cleaned up.